lib/services/local/users.go

```diff
@@ -1379,6 +1379,45 @@ func (s *IdentityService) GetMFADevices(ctx context.Context, user string, withSe
 		return nil, trace.BadParameter("missing parameter user")
 	}
 
+	// get normal MFA devices and SSO mfa device concurrently, returning the first error we get.
+	errC := make(chan error)
+
+	var devices []*types.MFADevice
+	go func() {
+		var err error
+		devices, err = s.getMFADevices(ctx, user, withSecrets)
+		if err != nil {
+			errC <- trace.Wrap(err)
+			return
+		}
+		errC <- nil
+	}()
+
+	var ssoDev *types.MFADevice
+	go func() {
+		var err error
+		ssoDev, err = s.getSSOMFADevice(ctx, user)
+		if !trace.IsNotFound(err) {
+			errC <- trace.Wrap(err)
+			return
+		}
+		errC <- nil
+	}()
+
+	for i := 0; i < 2; i++ {
+		if err := <-errC; err != nil {
+			return nil, trace.Wrap(err)
```

Comment:
Please wait for all goroutines to complete before returning.

Also note that, because you have an unbuferred channel, one of them is going to hang if the other errors.

Reply:
🤦 👍